CMakeLists.txt

```diff
@@ -5,16 +5,18 @@ enable_testing()
 list(APPEND CMAKE_MODULE_PATH ${PROJECT_SOURCE_DIR}/cmake)
```

Comment:
We should update the minimum_require to 3.12 since `add_compile_definitions ` is introduced in 3.12: https://cmake.org/cmake/help/latest/command/add_compile_definitions.html